tests/opentracing_instrumentation/test_traced_function_decorator.py

```diff
@@ -104,99 +260,76 @@ def run():
 
     @gen_test
     def test_decorator_with_name(self):
-        class SomeClient(object):
-            @traced_function(name='func2_modified')
-            @gen.coroutine
-            def func2(self, param1):
-                assert param1 == 123
-                raise tornado.gen.Return('oh yeah')
-
-        s = SomeClient()
+
         parent = opentracing.tracer.start_span('hello')
 
         @gen.coroutine
         def run():
             # verify start_span is called with overridden function name
-            child = mock.MagicMock()
-            with patch_object(opentracing.tracer, 'start_span',
-                              return_value=child) as start_child:
-                child.set_tag = mock.MagicMock()
-                r = yield s.func2(123)
-                assert r == 'oh yeah'
-                start_child.assert_called_once_with(
-                    operation_name='func2_modified',  # overridden name
-                    child_of=parent.context,
-                    tags=None)
-                assert child.set_tag.call_count == 0
+            for func in ('regular_with_name', 'coro_with_name', ):
+                child = mock.MagicMock()
+                with patch_object(opentracing.tracer, 'start_span',
+                                  return_value=child) as start_child:
+                    child.set_tag = mock.MagicMock()
+                    r = yield self.call(func, 123)
+                    assert r == 'oh yeah'
+                    start_child.assert_called_once_with(
+                        operation_name='some_name',  # overridden name
+                        child_of=parent.context,
+                        tags=None)
+                    assert child.set_tag.call_count == 0
 
-            raise tornado.gen.Return(1)
+                raise tornado.gen.Return(1)
```

Comment:
Hello @condorcet,

It looks like this `raise` will break the `for` loop at the first iteration and `'coro_with_name'` will not be processed.
Am I missing something?

Reply:
Nice catch, thank you @Jamim ! I've fixed it.